rust/src/treefile.rs

```diff
@@ -1873,6 +1889,7 @@ impl Treefile {
                 .as_ref()
                 .and_then(|m| m.install.as_ref().map(|i| !i.is_empty()))
                 .unwrap_or_default()
+            || self.parsed.derive.remove_kernel
```

Comment:
Additionally, we should clear it as part of `remove_all_overrides`. And we probably want to neuter it in the `validate_for_containers` hack so it works fine in containers too.